README.md

```diff
@@ -134,7 +134,194 @@ TODO
+element.  Each struct element is encoded as if followed by a `Start struct`
+field key (i.e. it is implied), and terminates with an `End struct` field key
+(specifically, the byte 0x04).
+
+```golang
+type Item struct {
+	Number int
+}
+type List struct {
+	MyList []Item
+}
+
+list := List{[]Item{Item{1}, Item{3}}}
+
+bz, err := wire.MarshalBinary(list)
+if err != nil { ... }
+
+// dump bz:
+// BINARY      HEX   NOTE
+// b0000 0011  0x03  "Start struct" for `List`
+// b0000 1110  0x14  field number and type for `MyList` (1, "List")
+// b0000 0011  0x03  type of element of `MyList`, a struct
+// b0000 0010  0x02  length of list, uvarint for 2
+//                   "Start struct" for `Item` #0 (implied)
+// b0000 1000  0x08  field number and type for Number (1, "Varint)
+// b0000 0010  0x02  field value varint for 1
+// b0000 0100  0x04  "End struct" for `Item` #0
+//                   "Start struct" for `Item` #0 (implied)
+// b0000 1000  0x08  field number and type for Number (1, "Varint)
+// b0000 0110  0x06  field value varint for 3
+// b0000 0100  0x04  "End struct" for `Item` #1
+// b0000 0100  0x04  "End struct" for `List`
+```
+
+A list of `n` elements where the elements are list of structs is encoded by the
+bytes `0x06 0x03` followed by the uvarint encoding of `n`, followed by the
+binary encoding of each element, which starts with the uvarint encoding of `m`
+(the size of the first child list).  Each struct element is encoded as if
+followed by a `Start struct` field key, as in the previous example.
+
+```golang
+type Item struct {
+	Number int
+}
+type List []Item
+type ListOfLists struct {
+	MyLists []List
+}
+
+llist := ListOfLists{List{[]Item{Item{1}, Item{3}}}}
+
+bz, err := wire.MarshalBinary(llist)
+if err != nil { ... }
+
+// dump bz:
+// BINARY      HEX   NOTE
+// b0000 0011  0x03  "Start struct" for `ListOfLists`
+// b0000 1110  0x14  field number and type for `MyLists` (1, "[]List")
+// b0000 0110  0x06  type of element of `MyLists`, a list
+// b0000 0011  0x03  type of element of `MyLists[0]`, a struct (continued type-bytes)
+// b0000 0001  0x01  length of list, uvarint for 1
+//                   "Start struct" for `List` #0 (implied)
+// b0000 1110  0x14  field number and type for MyList (1, "List")
+// b0000 0011  0x03  type of list element, struct
+// b0000 0010  0x02  length of list, uvarint for 2
+//                   "Start struct" for `Item` #0 (implied)
+// b0000 1000  0x08  field number and type for Number (1, "Varint)
+// b0000 0010  0x02  field value varint for 1
+// b0000 0100  0x04  "End struct" for `Item` #0
+//                   "Start struct" for `Item` #0 (implied)
+// b0000 1000  0x08  field number and type for Number (1, "Varint)
+// b0000 0110  0x06  field value varint for 3
+// b0000 0100  0x04  "End struct" for `Item` #1
+// b0000 0100  0x04  "End struct" for `List` #0
+// b0000 0100  0x04  "End struct" for `ListOfLists`
+```
+
+Unlike fields of a struct where nil pointers are denoted by the absence of
```

Comment:
> Unlike fields of a struct where nil pointers are denoted by the absence of encoding, elements of a list are encoded without an index or key.

What part about "nil pointers of a struct" and "elements of a list" is related?

Reply:
they're both conceptually items of a set.  In Golang there is is a distinction, but in some languages like Javascript that distinction is blurred... so I think it's nice to help explain the encoding by comparing how they are different.

List encoding could be more similar to struct encoding, by prefixing each element with a key that is the index number.  But instead, the encoding specified here is more compact for dense lists because the index number is not being encoded.  In the future, for sparse lists we could support >1 nil-items-skipping (the current spec only allows encoding 1 nil item at a time), which could be more compact than struct-like-list encoding still, because we're still not encoding the item keys.  But anyhow we should delay deciding on this (I think) because we don't need this feature, and, it's tricky... as is, the current spec makes the byte-length of the input be more-or-less representative of the amount of memory it takes to decode it.  A 1KB go-wire binary blob shouldn't decode into a 1GB object in memory.... Whereas with sparse encoding, it might.  So, something to think about.